blessed/sequences.py

```diff
@@ -315,6 +315,29 @@ def center(self, width, fillchar=u' '):
             (max(0.0, math.ceil(split))) / float(len(fillchar)))
         return u''.join((leftside, self, rightside))
 
+    def truncate(self, width):
+        """
+        Truncate a string in a sequence-aware manner.
+
+        Any printable characters beyond ``width`` are removed, while all
+        sequences remain in place.
+
+        :arg int width: The printable width to truncate the string to.
+        :rtype: str
+        :returns: String truncated to at most ``width`` printable characters.
+        """
+        output = ""
+        current_width = 0
+        for text, cap in iter_parse(self._term, self):
+            if not cap:
+                char_width = wcwidth(text)
+                current_width += char_width
```

Comment:
Could probably do `current_width += wcwidth(text)` since char_width isn't used anywhere else

Reply:
Fixed in 07014ba